packages/apps/slack/src/services/slack.ts

```diff
@@ -0,0 +1,159 @@
+import fs from "fs";
+import axios from "axios";
+import { Readable } from "stream";
+import { OAuthService, Organisation } from "@ocular/ocular";
+import {
+  IndexableDocument,
+  TransactionBaseService,
+  Logger,
+  AppNameDefinitions,
+} from "@ocular/types";
+import { ConfigModule } from "@ocular/ocular/src/types";
+
+interface Config {
+  headers: {
+    Authorization: string;
+    Accept: string;
+  };
+}
+
+export default class SlackService extends TransactionBaseService {
+  protected oauthService_: OAuthService;
+  protected logger_: Logger;
+  protected container_: ConfigModule;
+
+  constructor(container) {
+    super(arguments[0]);
+    this.oauthService_ = container.oauthService;
+    this.logger_ = container.logger;
+    this.container_ = container;
+  }
+
+  async getSlackData(org: Organisation) {
+    return Readable.from(this.getSlackChannelsAndConversations(org));
+  }
+
+  async *getSlackChannelsAndConversations(
+    org: Organisation
+  ): AsyncGenerator<IndexableDocument[]> {
+    this.logger_.info(`Starting oculation of Slack for ${org.id} organisation`);
+
+    // Get Slack OAuth for the organisation
+    const oauth = await this.oauthService_.retrieve({
+      id: org.id,
+      app_name: AppNameDefinitions.SLACK,
+    });
+
+    if (!oauth) {
+      this.logger_.error(`No Slack OAuth found for ${org.id} organisation`);
+      return;
+    }
+
+    const config: Config = {
+      headers: {
+        Authorization: `Bearer ${oauth.token}`,
+        Accept: "application/json",
+      },
+    };
+
+    let documents: IndexableDocument[] = [];
+
+    try {
+      const slackChannels = await this.fetchSlackChannels(config)
+
+      for (const channel of slackChannels) {
```

Comment:
Line 64 and 65 looks good!!

On line 66: I am thinking of creating documents on the basis of threads and not individual methods.

For example: 
- Iterate through the conversations
   if conversation.thread has not been seen
      - find all messages in that thread.
      - Create an IndexableDocument ( tittle = name of first messge in thread, Sections = [Messages]
      - mark thread seen.